pkg/storage/stores/indexshipper/compactor/compactor.go

```diff
@@ -533,11 +581,20 @@ func (c *Compactor) RegisterIndexCompactor(indexType string, indexCompactor Inde
 }
 
 func (c *Compactor) RunCompaction(ctx context.Context, applyRetention bool) error {
+	c.compactionMtx.Lock()
```

Comment:
Ensure `RunCompaction` and `RunSizeBasedCompaction` can't be run at the same time